stdlib/src/builtin/string.mojo

```diff
@@ -1406,6 +1410,35 @@ struct String(
             substr._strref_dangerous(), start=start
         )
 
+    fn isspace(self) -> Bool:
+        """Determines whether the given String is a python
+        whitespace String. This corresponds to Python's
+        [universal separators](
+            https://docs.python.org/3/library/stdtypes.html#str.splitlines)
+        `" \\t\\n\\r\\f\\v\\x1c\\x1e\\x85\\u2028\\u2029"`.
+
+        Returns:
+            True if the String is one of the whitespace characters
+                listed above, otherwise False."""
+        # TODO add line and paragraph separator as stringliteral
+        # once unicode escape secuences are accepted
+        # Unicode Line Separator: \ u2028
+        # Unicode Paragraph Separator: \ u2029
+        # 0 is to build a String with null terminator
+        return self in List[String](
```

Comment:
The perf implications of this still bother me. The cpython implementation checks for the length first, and dispatches a lookup table based solution for single character strings. I think this is definitely something we should do (see my other comment).

Also, the cpython implementation actually returns true if _all_ elements in a string are spaces, i.e. not just for single element strings. This was a surprise to me, but we should follow it for now, and we can think about different APIs later.